File: packages/neptune_flutter_ui/lib/src/widgets/neptune_stat_card.dart
// © 2026 Neptune.Fintech (neptune.ly) · Neptune Odyssey Community License v1.0

import 'package:flutter/material.dart';

import '../theme/extensions.dart';
import '../theme/neptune_theme.dart';
import 'neptune_identity_surfaces.dart';

/// A compact metric tile (web `<npt-stat-card>`): a label, a big tabular value
/// with an optional unit, a signed delta coloured success/error, and an
/// optional chart slot (e.g. a sparkline). Theme-only, RTL-safe.
class NeptuneStatCard extends StatelessWidget {
  final String label;
  final String value;
  final String? unit;
  final String? delta;
  final Widget? chart;

  const NeptuneStatCard({
    super.key,
    required this.label,
    required this.value,
    this.unit,
    this.delta,
    this.chart,
  });

  @override
  Widget build(BuildContext context) {
    final scheme = Theme.of(context).colorScheme;
    final shape = Theme.of(context).extension<NptShape>()!;
    final npt = Theme.of(context).extension<NptColors>()!;
    final text = Theme.of(context).textTheme;
    final money = NeptuneTheme.moneyStyle(context, base: text.titleLarge)
        .copyWith(color: scheme.onSurface, fontWeight: FontWeight.w700);
    final isDown = (delta ?? '').trimLeft().startsWith('-');
    final deltaColor = isDown ? scheme.error : npt.success;

    return Container(
      decoration: BoxDecoration(color: scheme.surfaceContainer, borderRadius: shape.rLg),
      padding: const EdgeInsetsDirectional.all(16),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        mainAxisSize: MainAxisSize.min,
        children: [
          // The Odyssey eyebrow — uppercase, tracked, display face.
          NeptuneEyebrow(label),
          const SizedBox(height: 6),
          Row(
            crossAxisAlignment: CrossAxisAlignment.baseline,
            textBaseline: TextBaseline.alphabetic,
            children: [
              Flexible(
                child: Text(value, style: money, maxLines: 1, overflow: TextOverflow.ellipsis),
              ),
              if (unit != null) ...[
                const SizedBox(width: 6),
                Text(unit!, style: text.labelLarge?.copyWith(color: scheme.onSurfaceVariant)),
              ],
            ],
          ),
          if (delta != null) ...[
            const SizedBox(height: 6),
            Text(
              delta!,
              style: text.labelMedium?.copyWith(color: deltaColor, fontWeight: FontWeight.w700),
            ),
          ],
          if (chart != null) ...[const SizedBox(height: 10), chart!],
        ],
      ),
    );
  }
}

File: packages/neptune_kmp_ui/odyssey-compose-ui/src/commonMain/kotlin/ly/neptune/odyssey/ui/components/NeptuneStatCard.kt
// © 2026 Neptune.Fintech (neptune.ly) · Neptune Odyssey Community License v1.0
//
// The compact metric tile: an Odyssey eyebrow label, a big tabular money
// value with an optional unit on the baseline, a signed delta coloured
// success/error, and an optional chart slot (e.g. a sparkline). Web
// counterpart: `<npt-stat-card>` · Flutter: neptune_stat_card.dart.
// Theme-only, RTL-safe.

package ly.neptune.odyssey.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import ly.neptune.odyssey.ui.identity.NeptuneEyebrow
import ly.neptune.odyssey.ui.theme.NeptuneTheme

/**
 * A compact metric tile (web `<npt-stat-card>`): [label] as the eyebrow,
 * [value] in the brand money style (tabular figures, w700) with an optional
 * [unit] on the baseline, an optional signed [delta] coloured success (or
 * error when it starts with `-`), and an optional [chart] slot below.
 *
 * Web counterpart: `<npt-stat-card>` · Flutter: `NeptuneStatCard`.
 */
@Composable
public fun NeptuneStatCard(
    label: String,
    value: String,
    modifier: Modifier = Modifier,
    unit: String? = null,
    delta: String? = null,
    chart: (@Composable () -> Unit)? = null,
) {
    val scheme = MaterialTheme.colorScheme
    val typography = MaterialTheme.typography
    val money = NeptuneTheme.moneyStyle(base = typography.titleLarge)
        .copy(color = scheme.onSurface, fontWeight = FontWeight.W700)
    val isDown = (delta ?: "").trimStart().startsWith("-")
    val deltaColor = if (isDown) scheme.error else NeptuneTheme.colors.success

    Column(
        modifier
            .background(scheme.surfaceContainer, NeptuneTheme.shape.rLg)
            .padding(16.dp),
    ) {
        NeptuneEyebrow(label)
        Spacer(Modifier.height(6.dp))
        Row {
            Text(
                NeptuneTheme.formatDigits(value),
                style = money,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis,
                modifier = Modifier.weight(1f, fill = false).alignByBaseline(),
            )
            if (unit != null) {
                Spacer(Modifier.width(6.dp))
                Text(
                    unit,
                    style = typography.labelLarge.copy(color = scheme.onSurfaceVariant),
                    modifier = Modifier.alignByBaseline(),
                )
            }
        }
        if (delta != null) {
            Spacer(Modifier.height(6.dp))
            Text(
                NeptuneTheme.formatDigits(delta),
                style = typography.labelMedium.copy(color = deltaColor, fontWeight = FontWeight.W700),
            )
        }
        if (chart != null) {
            Spacer(Modifier.height(10.dp))
            chart()
        }
    }
}
